src/common/vector_operations/comparison_operators.cpp

```diff
@@ -182,6 +182,10 @@ static void NestedComparisonExecutor(Vector &left, Vector &right, Vector &result
 	if (!leftv.validity.AllValid() || !rightv.validity.AllValid()) {
 		ComparesNotNull(leftv, rightv, result_validity, count);
 	}
+	ValidityMask copy;
+	copy.SetAllValid(count);
+	copy.Copy(result_validity, count);
```

Comment:
Do we need `SetAllValid`, if we are immediately copying `result_validity` into it?
```cpp
	inline void Copy(const TemplatedValidityMask &other, idx_t count) {
		target_count = count;
		if (other.AllValid()) {
			validity_data = nullptr;
			validity_mask = nullptr;
		} else {
			validity_data = make_buffer<ValidityBuffer>(other.validity_mask, count);
			validity_mask = validity_data->owned_data.get();
		}
	}
```